_plugins/jekyll_regex_match.rb

```
@@ -20,6 +20,35 @@ def regex_match_once(input_str, regex_str)
+            # Track cache size
+            @cache_size ||= 0
+            @cache_size += 1 unless item['cached_dates']
+
+            dates = item['cached_dates'] ||= item['date'].scan(regex).map(&:to_s) 
+            dates.include?(year)  
+          end
+          rescue RegexpError => e
+            # Handle invalid regex pattern
+            []
+          end
+        end
```

rubocop

[convention] Assignment Branch Condition size for filter_items_by_year is too high. [<10, 18, 16> 26.08/23]

(Metrics/AbcSize)